src/datepicker/datepicker.spec.ts

```diff
@@ -541,6 +572,143 @@ describe('ngb-datepicker', () => {
       expect(getYearSelect(fixture.nativeElement).value).toBe(`${today.getFullYear()}`);
     });
 
+    it('should correctly navigate with keyboard', () => {
```

Comment:
This test is quite big, and if one condition fails we would have hard time figuring out what failed exactly. 

Could you split it into multiple small tests? (ideally one test for one navigation key)?